hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1439,8 +1439,12 @@ int QuectelNcpClient::registerNet() {
             r = CHECK_PARSER(respNwMode.readResult());
             CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_UNKNOWN);
         #if PLATFORM_ID == PLATFORM_MSOM
-            if (nwScanMode != 0) {
-                CHECK_PARSER(parser_.execCommand("AT+QCFG=\"nwscanmode\",0,1")); // AUTO
+            // M404/BG95M5 should be LTEM only, ie scan mode 3
+            // M524/EG91EX should be AUTO (both LTEM and 2G), ie scan mode 0
+            int desiredNwScanMode = (ncpId() == PLATFORM_NCP_QUECTEL_BG95_M5) ? 3 : 0;
```

Comment:
This check will never test for M524 because it is not PLATFORM_NCP_QUECTEL_BG96 || PLATFORM_NCP_QUECTEL_BG95_M5, but rather PLATFORM_NCP_QUECTEL_EG91_EX.  I think we might need to add a  `if (isQuecCat1Device()) { }` block if we want to ensure `nwscanmode` is on the default of 0 for those devices.  For this block you can simple check `if (nwScanMode != 3) {` and set to 3 if not.

Reply:
Good point, I didnt realize this wasnt checking for EG91. So M524 NWSCAN is not being set, presumably being left as default (ie enable 2g)?

This does disable 2g for specifically M404/BG95M5, and leaves it enabled for BG95M6 devices (which I dont think there is any aside from a few samples?)

Reply:
Yeah I think EG91_EX will be left at defaults, 0.